qcengine/programs/openmm.py

```diff
@@ -0,0 +1,194 @@
+"""
+Calls the OpenMM executable.
+
+Requires RDKit
+"""
+import json
+import os
+from typing import Dict
+
+from qcelemental.models import Provenance, Result
+from qcelemental.util import which_import
+
+from .model import ProgramHarness
+from ..exceptions import InputError, RandomError, ResourceError, UnknownError
+from ..util import execute, popen, temporary_directory
+from ..units import ureg
+
+from .rdkit import RDKitHarness
+
+
+class OpenMMHarness(ProgramHarness):
+
+    # TODO: verify correctness of these default params
```

Comment:
`thread_safe` is True/False depending if I can evaluate two instances concurrent in the same Python interpreter. This is fine only if a program does not use global memory and uses something more than PID unique files. This is typically used by our orchestration layer to help determine scheduling.

Reply:
Ah, I see. I'm actually not certain how OpenMM behaves in this regard, as I've never tried to run e.g. two `Simulation` instances in different threads on the same interpreter. I'm willing to bet the answer is `False` here. If anyone knows otherwise, please let us know.

Reply:
OpenMM should be thread-safe provided you are using separate `Context` objects in the threads you are trying to perform operations on concurrently.

Reply:
Great! Switching `thread_safe` to `True`, and adding this proviso as an inline comment for reference.